src/sources/util/http.rs

```diff
@@ -49,6 +50,55 @@ impl fmt::Debug for RejectShuttingDown {
 }
 impl warp::reject::Reject for RejectShuttingDown {}
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct HttpSourceAuthConfig {
+    pub username: String,
```

Comment:
Could this hold some `Authorization` variant? I expect we'll see a request for [bearer](https://docs.rs/headers/0.3.2/headers/struct.Authorization.html#method.bearer) soon 😏

Reply:
It's should be possible to add bearer in the future without problems, we will change the current struct to enum like in [sinks/util/http](https://github.com/timberio/vector/blob/1e95458540596b7e0058684afc8ac97caa84e992/src/sinks/util/http.rs#L375-L380):
```rust
#[derive(Deserialize, Serialize, Clone, Debug)]
#[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
pub enum HttpSourceAuthConfig {
    Basic { username: String, password: String },
    Bearer { token: String },
}
```